api/src/main/java/org/openmrs/module/pihmalawi/validator/impl/YendaNafePatientRegistrationValidatorImpl.java

```diff
@@ -0,0 +1,75 @@
+package org.openmrs.module.pihmalawi.validator.impl;
+
+import org.openmrs.Location;
+import org.openmrs.PatientIdentifier;
+import org.openmrs.api.LocationService;
+import org.openmrs.module.pihmalawi.models.YendaNafePatientRegistrationModel;
+import org.openmrs.module.pihmalawi.patient.ExtendedPatientService;
+import org.openmrs.module.pihmalawi.validator.YendaNafePatientRegistrationValidator;
+import org.openmrs.module.reporting.common.DateUtil;
+import org.springframework.stereotype.Component;
+
+import java.util.Date;
+
+/**
+ *  Class implements validation adding patient from medic mobile data model for patient registration
+ */
+@Component
+public class YendaNafePatientRegistrationValidatorImpl implements YendaNafePatientRegistrationValidator {
+    public String validateRegistrationModel(YendaNafePatientRegistrationModel yendaNafePatientRequestBody, ExtendedPatientService extendedPatientService, LocationService locationService)
+    {
+        PatientIdentifier patientIdentifierInDB = extendedPatientService.getPatientIdentifierByIdentifier(yendaNafePatientRequestBody._id);
+        if(patientIdentifierInDB != null)
+        {
+            return "_id "+ yendaNafePatientRequestBody._id +" already exist in the system. _id has to be unique";
+        }
+        if(yendaNafePatientRequestBody.name.equals(""))
+        {
+            return "Name of patient not given.";
+        }
+
+        String[] parts = yendaNafePatientRequestBody.name.split(" ", 2);
+        if(parts.length == 1)
+        {
+            return "Name is incomplete. Please provide first name and last name.";
+        }
+        String sex = yendaNafePatientRequestBody.sex.toLowerCase().trim();
```

Comment:
Might want a null-check here?

Reply:
@mseaton For null checks, I implemented in the default constructor to initialize all strings to empty string("")  so i was checking that for every property.  Come to think of it, null for every field would be better than empty string. So i will update the constructor and code to validate the fields for nulls than the empty string.